wrappers/internal/service_unit_gen.go

```diff
@@ -0,0 +1,335 @@
+				logger.Noticef("Cannot get 'name' attribute of dbus slot %q: %v", slot.Name, err)
+			}
+		}
+	}
+
+	wrapperData := struct {
+		App *snap.AppInfo
+
+		Restart                  string
+		WorkingDir               string
+		StopTimeout              time.Duration
+		StartTimeout             time.Duration
+		ServicesTarget           string
+		PrerequisiteTarget       string
+		MountUnit                string
+		Remain                   string
+		KillMode                 string
+		KillSignal               string
+		OOMAdjustScore           int
+		BusName                  string
+		Before                   []string
+		After                    []string
+		InterfaceServiceSnippets string
+		SliceUnit                string
+		LogNamespace             string
+
+		Home    string
+		EnvVars string
+
+		CoreMountedSnapdSnapDep []string
+	}{
+		App: appInfo,
+
+		InterfaceServiceSnippets: ifaceSpecifiedServiceSnippet,
+
+		Restart:        restartCond,
+		StopTimeout:    serviceStopTimeout(appInfo),
+		StartTimeout:   time.Duration(appInfo.StartTimeout),
+		Remain:         remain,
+		KillMode:       killMode,
+		KillSignal:     appInfo.StopMode.KillSignal(),
+		OOMAdjustScore: oomAdjustScore,
+		BusName:        busName,
+
+		Before: generateServiceNames(appInfo.Snap, appInfo.Before),
+		After:  generateServiceNames(appInfo.Snap, appInfo.After),
+
+		// systemd runs as PID 1 so %h will not work.
+		Home: "/root",
+	}
+	switch appInfo.DaemonScope {
+	case snap.SystemDaemon:
+		wrapperData.ServicesTarget = systemd.ServicesTarget
+		wrapperData.PrerequisiteTarget = systemd.PrerequisiteTarget
+		wrapperData.MountUnit = filepath.Base(systemd.MountUnitPath(appInfo.Snap.MountDir()))
+		wrapperData.WorkingDir = appInfo.Snap.DataDir()
+		wrapperData.After = append(wrapperData.After, "snapd.apparmor.service")
+	case snap.UserDaemon:
+		wrapperData.ServicesTarget = systemd.UserServicesTarget
+		// FIXME: ideally use UserDataDir("%h"), but then the
+		// unit fails if the directory doesn't exist.
+		wrapperData.WorkingDir = appInfo.Snap.DataDir()
+	default:
+		panic("unknown snap.DaemonScope")
+	}
+
+	// check the quota group slice
+	if opts.QuotaGroup != nil {
+		wrapperData.SliceUnit = opts.QuotaGroup.SliceFileName()
+		if opts.QuotaGroup.JournalQuotaSet() {
+			wrapperData.LogNamespace = opts.QuotaGroup.JournalNamespaceName()
+		}
+	}
+
+	// Add extra "After" targets
+	if wrapperData.PrerequisiteTarget != "" {
+		wrapperData.After = append([]string{wrapperData.PrerequisiteTarget}, wrapperData.After...)
+	}
+	if wrapperData.MountUnit != "" {
+		wrapperData.After = append([]string{wrapperData.MountUnit}, wrapperData.After...)
+	}
+
+	if opts.RequireMountedSnapdSnap {
+		// on core 18+ systems, the snapd tooling is exported
+		// into the host system via a special mount unit, which
+		// also adds an implicit dependency on the snapd snap
+		// mount thus /usr/bin/snap points
+		wrapperData.CoreMountedSnapdSnapDep = []string{SnapdToolingMountUnit}
```

Comment:
in this refactoring this is a bit strange, it would make more sense to make the option the optional path that we need to use and keep SnapdToolingMountUnit defined on the outside only

Reply:
This makes a lot of sense - i'll do that modification!